README.rst

```diff
@@ -271,6 +271,19 @@ __ https://docs.scale.com/reference/delete-task-unique-id
     task.clear_unique_id()
 
 
+Set A Task's Metadata
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Set a given task's `metadata`. Check out `Scale's API documentation`__ for more information.
+
+__ https://docs.scale.com/reference/set-metadata
+
+.. code-block :: python
+
+    # set metadata on a task to be the contents of new_metadata
+    new_metadata = {'metadata': {'myKey': 'myValue'}}
```

Comment:
I think we append metadata in the method? So users should only pass `{'myKey': 'myValue'}`?

Reply:
Also should we govern that in the method? When they check the API guide they may think they need to pass metadata as well, but appending that ourselves is probably a better user experience.

Reply:
Possible oversight here, but the current implementation of the API endpoint does expect that the customer will provide the metadata in the format of `{"metadata": {"key": "value"}}. 

Should we revise the endpoint and also update the SDK to reflect this change?

Reply:
Ah.. yeah, we probably should - sorry I missed that nuance.

Reply:
No matter what we choose, I think we should be consistent all over the place. 
If endpoint expects a `metadata` as the main key then we should:
- expect user to follow that format
- is this enforced in API? If not, we may also govern that in SDK as an additional measure. metadata key to be there, be an object? and be the only key in the main object?
- Or we should update the API just to accept any object and push the same under metadata. (so users does not need to add extra metadata field) -> I'm pro to this.

- We may not add extra metadata in the method: https://github.com/scaleapi/scaleapi-python-client/pull/56/files#diff-09863685c9b788d64d266a4468f8e7f893e1d1c7773c9e20c6757b70fd378608R130 just expect user to do that if API expects. `payload = {"metadata": metadata}`

Reply:
The API currently does enforce having the "metadata" key

I'll revise the logic in the API and then revise the behavior in the SDK today so that they are consistent

Reply:
PR created for change to API endpoint: https://github.com/scaleapi/scaleapi/pull/47725